salt/modules/mac_user.py

```diff
@@ -140,6 +142,9 @@ def delete(name, *args):
         raise SaltInvocationError('Username cannot contain whitespace')
     if not info(name):
         return True
+
+    # remove home directory from filesystem
+    __salt__['file.remove'](info(name)['home'])
```

Comment:
Are we sure that we want to do this? What's the default on OS X when a user is removed?